src/components/LinkManager/LinkManager.tsx

```diff
@@ -0,0 +1,135 @@
+// client component 선언
+"use client";
+// react
+import { ChangeEvent, use, useEffect, useRef, useState } from "react";
+// fetch API
+import { END_POINT } from "@/src/constants";
+// component
+import LinkList from "./LinkList/LinkList";
+import FavoriteButtonList from "./FavoriteButtonList/FavoriteButtonList";
+import LinkSearchBar from "./LinkSearchBar/LinkSearchBar";
+import LinkEmptyCase from "./LinkList/LinkEmptyCase/LinkEmptyCase";
+import AddLinkBar from "./AddLinkBar/AddLinkBar";
+import UserProfile from "./UserProfile/UserProfile";
+// types
+import {
+  Favorite,
+  Link as LinkType,
+  SharedPageTypes,
+} from "@/src/constants/types";
+// utils
+import getFormattedLinks from "@/src/utils/getFormattedLinks";
+import useIntersectionObserver from "@/src/hooks/useIntersectionObserver";
+// style sheet
+import styles from "./LinkManager.module.scss";
+
+interface LinkManager {
+  links: LinkType[];
+  favorites?: Favorite[];
+  userProfile?: SharedPageTypes;
```

Comment:
아래처럼 되는게 일반적일 것 같아서 지금은 어색하게 느껴지네요!
```suggestion
  userProfile?: UserProfile;
```